arm7/lib/src/SND_util.c

```diff
@@ -0,0 +1,109 @@
+#include "SND_util.h"
+
+#include "syscall.h"
+
+// TODO remove this extern once the static const definition of it is here
+extern s8 sLfoSinTable[0x21];
+
+u16 SND_CalcTimer(int timer, int pitch)
+{
+    int octave = 0;
+    int pitch_normalized = -pitch;
+
+    while (pitch_normalized < 0)
+    {
+        octave--;
+        pitch_normalized += 768;
+    }
+
+    while (pitch_normalized >= 768)
+    {
+        octave++;
+        pitch_normalized -= 768;
+    }
+
+    u64 result = SVC_GetPitchTable(pitch_normalized);
+
+    result += 0x10000;
+    result *= timer;
+
+    int shift = octave - 16;
+
+    if (shift <= 0)
+    {
+        shift = -shift;
+        result >>= shift;
+    }
+    else if (shift < 32)
+    {
+        // what ???
+        u64 tmp = result & ~0uLL << (32 - shift);
```

Comment:
wut

Reply:
Yeah, I think that was one of the things that I genuinely did not understand when I was decompiling it.